frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -761,15 +784,73 @@ export const CRDCardRow = connect(crdCardRowStateToProps)((props: CRDCardRowProp
   );
 });
 
+const InitializationResourceAlert: React.SFC<InitializationResourceAlertProps> = (props) => {
+  const { initializationResource, csv } = props;
+
+  const initializationResourceKind = initializationResource?.kind;
+  const initializationResourceDefinition = props.csv.spec?.customresourcedefinitions?.owned?.find(
+    (api) => api.kind === initializationResourceKind,
+  );
+  const initializationResourceModelName = initializationResourceDefinition?.name;
+  const { group: initializationResourceGroup } = groupVersionFor(
+    initializationResource?.apiVersion,
+  );
+  const { findModel } = useModelFinder();
+  const model = findModel(
+    initializationResourceGroup,
+    initializationResourceModelName.split('.').shift(),
+  );
+  // Check if the CR is already present - only checks for the specific name/ns in the initialization-resource
+  const [customResource, customResourceLoaded] = useK8sGet<K8sResourceCommon>(
+    model,
+    initializationResource?.metadata.name,
+    initializationResource?.metadata.namespace || csv.metadata?.namespace,
+  );
+  const canCreateCustomResource = useAccessReview({
+    group: initializationResourceGroup,
+    resource: initializationResourceModelName.split('.').shift(),
+    verb: 'create',
+  });
+
+  if (!customResource && customResourceLoaded && canCreateCustomResource) {
+    return (
+      <Alert
+        isInline
+        className="co-alert"
+        variant="warning"
+        title={`${initializationResourceKind} Required`}
+      >
+        <p>Create a {initializationResourceKind} instance to use this operator.</p>
+        <CreateInitializationResourceButton
+          obj={props.csv}
+          targetNamespace={initializationResource?.metadata.namespace || csv.metadata?.namespace}
+        />
+      </Alert>
+    );
+  }
+  return null;
+};
+
 export const ClusterServiceVersionDetails: React.SFC<ClusterServiceVersionDetailsProps> = (
   props,
 ) => {
   const { spec, metadata, status } = props.obj;
   const {
     'marketplace.openshift.io/support-workflow': marketplaceSupportWorkflow,
     'olm.targetNamespaces': olmTargetNamespaces = '',
+    'operatorframework.io/initialization-resource': initializationResourceJSON,
   } = metadata.annotations || {};
 
+  let initializationResource = null;
+  if (initializationResourceJSON) {
+    try {
+      initializationResource = JSON.parse(initializationResourceJSON);
+    } catch (error) {
+      const errorMsg = error.message;
+      errorModal({ errorMsg });
```

Comment:
I wouldn't show a modal because the user can't do anything. Just use `console.error`